sys/net/application_layer/nanocoap/sock.c

```diff
@@ -333,9 +333,15 @@ ssize_t _sock_put_post(nanocoap_sock_t *sock, const char *path, unsigned code,
         .iov_len  = max_len,
     };
 
-    pktpos += coap_build_hdr(pkt.hdr, COAP_TYPE_CON, NULL, 0, code, _get_id());
+    pktpos += coap_build_hdr(pkt.hdr, type, NULL, 0, code, _get_id());
     pktpos += coap_opt_put_uri_path(pktpos, 0, path);
 
+    if (response == NULL && type == COAP_TYPE_NON) {
+        /* all responses (2.xx, 4.xx and 5.xx) are ignored */
```

Comment:
Is that a good default? Just because the client tolerates if the message doesn't get through it doesn't mean it's not interested in being informed of errors (or even in the notification that it did go through).

I'm aware that adding extra _noresponse variants is large API surface, but I guess that's what comes with having options and not having a multi-call (eg. builder-like) request description.

Reply:
Hm we could just ignore 2.xx responses since we can't tell them apart from no responses with the current API (will just return 0 on success and not wait for a response).

Reply:
Actually we will not wait for *any* response with NON and no response buffer already

https://github.com/RIOT-OS/RIOT/blob/master/sys/net/application_layer/nanocoap/sock.c#L157

So the no-response option just tells the server what the client already does anyway - any response is currently ignored.